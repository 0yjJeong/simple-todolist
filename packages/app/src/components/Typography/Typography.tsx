import { FC } from 'react';
import styled from 'styled-components';

export enum Size {
  TITLE,
  SUB_TITLE,
  CAPTION,
}

type Styles = {
  '@size': number;
  '@weight': string;
};

export const typoStyles: {
  [key in Size]: Styles;
} = {
  0: {
    '@size': 36,
    '@weight': '500',
  },
  1: {
    '@size': 24,
    '@weight': '500',
  },
  2: {
    '@size': 12,
    '@weight': '500',
  },
};

export type TypographyProps = {
  text: string;
  size: Size;
};

export const TypographyBlock = styled.p<Styles>`
  margin: 0;
  font-size: ${(props) => props['@size']}px;
  font-weight: ${(props) => props['@weight']};
`;

const Typography: FC<TypographyProps> = ({ text, size }) => {
  return <TypographyBlock {...typoStyles[size]}>{text}</TypographyBlock>;
};

export default Typography;
