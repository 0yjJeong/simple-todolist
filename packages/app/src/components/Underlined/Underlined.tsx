import { FC } from 'react';
import styled from 'styled-components';
import {
  typoStyles,
  TypographyBlock,
  TypographyProps,
} from '../Typography/Typography';

const Typo = styled(TypographyBlock)`
  display: inline-block;
  position: relative;

  &:before {
    content: '';
    display: block;
    height: 6px;
    position: absolute;
    width: 100%;
    bottom: 4px;
    background: #66ff91;
    z-index: -1;
  }
`;

const Underlined: FC<TypographyProps> = ({ text, ...rest }) => {
  return <Typo {...typoStyles[rest.size]}>{text}</Typo>;
};

export default Underlined;
