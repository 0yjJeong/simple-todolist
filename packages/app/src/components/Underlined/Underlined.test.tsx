import { render } from '@testing-library/react';
import 'jest-styled-components';
import Underlined from './Underlined';
import { Size } from '../Typography/Typography';

describe('<Underlined/>', () => {
  it('should', () => {
    const rendered = render(<Underlined text='a' size='subTitle' />);
    expect(rendered.container.firstChild).toMatchSnapshot();
    expect(rendered.container.firstChild).toHaveStyleRule('z-index', '-1', {
      modifier: '&:before',
    });
    expect(rendered.container.firstChild).toHaveStyleRule('display', 'block', {
      modifier: '&:before',
    });
    expect(rendered.container.firstChild).toHaveStyleRule('content', "''", {
      modifier: '&:before',
    });
  });
});
