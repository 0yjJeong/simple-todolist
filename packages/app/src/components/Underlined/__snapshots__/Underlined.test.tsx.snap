// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Underlined/> should 1`] = `
.c0 {
  margin: 0;
  font-size: 24px;
  font-weight: 500;
}

.c1 {
  display: inline-block;
  position: relative;
}

.c1:before {
  content: '';
  display: block;
  height: 6px;
  position: absolute;
  width: 100%;
  bottom: 4px;
  background: #66ff91;
  z-index: -1;
}

<p
  class="c0 c1"
>
  a
</p>
`;
